catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/NsiliEndpoint.java

```diff
@@ -121,10 +122,32 @@ public void setQuerySources(List<String> querySources) {
         this.querySources.clear();
         if (querySources != null) {
             this.querySources.addAll(querySources);
+
+            if (library != null) {
+                library.setQuerySources(querySources);
+            }
         }
+    }
 
-        if (library != null) {
-            library.setQuerySources(querySources);
+    @Override
+    public void addQuerySource(String sourceId) {
+        if (querySources != null && !querySources.contains(sourceId)) {
+            this.querySources.add(sourceId);
+
+            if (library != null) {
```

Comment:
see previous comment, also, i see this if-statement repeated three times, could be a good candidate for refactoring into a new method, eg:

```private void setLibraryQuerySources(List<String> querySources) {
  Optional.ofNullable(library) .ifPresent(l -> l.setQuerySources(querySources));
}```